### ISSUE TITLE
sched.stop() doesn't stop it right away

### ISSUE THREAD

**Issue by runsun** [mohayonao/web-audio-scheduler#8]
Some code like this:

`sched.stop( true )  // stop old song`
`sched.start( somefunc ) // start new one`

doesn't seem to stop the old song right away, resulting in overlapping of the last note of old song and the first note of the new song.  

The task is to pick a new song from a play list. Users might listen to one song, then immediately want to listen to a new one.

**Comment by mohayonao** [comment]
Perhaps, that behavior is correct.
`web-audio-scheduler` handles sequence events only.
So, your notes sequence and play like the below figure.

<img width="535" alt="seq" src="https://cloud.githubusercontent.com/assets/158075/18030302/eeb80cd4-6ceb-11e6-82a0-9ecb8a7b52cb.png">

If you want to remove playback notes immediately when stop the old sequence, you should write the code yourself. The below example code is an easy solution.

```js
sched.on("stop", () => {
  song.masterGain.disconnect(); // remove notes of old song
  song.masterGain = audioContext.createGain(); // master gain for notes of new song
});
```

**Comment by runsun** [comment]
Good to know there's something to try. Can you tell me what the `song` is here ? Tried to figure it out myself but no success ( I'm new to web audio and the web-audio-schedular  is my first encounter in this direction )

**Comment by mohayonao** [comment]
`song.masterGain` is an example. In this case, you should create a node for mixing song's all notes yourself.  like this

<img width="424" alt="graph" src="https://cloud.githubusercontent.com/assets/158075/18032398/96c61cd4-6d3e-11e6-8092-a7997497ca1c.png">

- yellow node (`masterGain`) are created in the first of a song.
  - `masterGain` is connected to the end point of audio graph (e.g. AudioDestinationNode).
  - this node is used to control all notes. (e.g. remove all nodes  for a song)
- green nodes (note of song) are created in a sequence schedule.
  - these nodes are connected to `masterGain`.

You can disconnect the `masterGain` to remove all notes easily.

<img width="424" alt="graph2" src="https://cloud.githubusercontent.com/assets/158075/18032482/e6d1138a-6d40-11e6-9745-7e5c88aeec03.png">

**Comment by runsun** [comment]
Thx for the graphic. It seems to be a clear explanation, but I don't know how to get it to work. I am using your *metronome* example to insert sequence like 

`ched.insert( time, ticktack, { frequency: .., duration: ... })` 

I read some web audio stuff about createOscillator and createGain, but, looking into the code of *web-audio-scheduler.js*, I couldn't find any OscillatorNode/GainNode creation.  So I have no idea how this masterGian approach can be applied to the metronome.

**Comment by mohayonao** [comment]
`web-audio-scheduler` doesn't create any audio nodes. it is your work.

**web-audio-scheduler**
- call callback functions at suitable timing for web audio

**you**
- make a callback function and insert it to `web-audio-scheduler`
- when call a callback function, create audio node for playing needed


I've rewritten metronome demo using master gain pattern. please refer it.

https://gist.github.com/mohayonao/6f7daddda053b975a705ce8437e434ce
- [L:27](https://gist.github.com/mohayonao/6f7daddda053b975a705ce8437e434ce#file-web-audio-scheduler-metronome-example-js-L27) - (note of song) connect to master gain
- [L:44-48](https://gist.github.com/mohayonao/6f7daddda053b975a705ce8437e434ce#file-web-audio-scheduler-metronome-example-js-L44-L48) - create master gain when `sched.start()`
- [L:50-53](https://gist.github.com/mohayonao/6f7daddda053b975a705ce8437e434ce#file-web-audio-scheduler-metronome-example-js-L50-L53) - reset master gain when `sched.stop()`

**Comment by runsun** [comment]
Thx. Btw, the `masterGain` needs to connect to `audioContext.destination`.  I'm trying to make it work without success (still overlapping). Will need more sometime to see if it is my code going wrong.

**Comment by mohayonao** [comment]
> Thx. Btw, the masterGain needs to connect to audioContext.destination

yes. my example code is wrong.

```js
sched.on("start", () => {
  masterGain = audioContext.createGain();
  masterGain.connect(audioContext.destination);
});
```

still overlapping? umm.. let me see your code?

**Comment by runsun** [comment]
Sure I can show you the code (a html file). Let me know how. Before we do that, the new metronome example produces an error inside the ticktack():

    amp.gain.setValueAtTime(0.5, t0);
    amp.gain.exponentialRampToValueAtTime(1e-6, t1);
    amp.connect(masterGain); // <-- connect to master gain, instead of AudioDestinationNode

in the console:

   Uncaught TypeError: Failed to execute 'connect' on 'AudioNode': No function was found that matched the signature provided.

The `masterGain` was shown to be `null` inside ticktack.

**Comment by mohayonao** [comment]
oooops. That error caused by timing of "start" event.
I fixed timing of 'start' event at v1.2.0. The new demo will be able to run.

thanks!

**Comment by runsun** [comment]
Seems to work fine so far. Thx for the help. I'll tidy up the test code (an html file) and make a pull request so it can be used as a future example.
